iOS_SDK/OneSignalSDK/Source/OneSignalLifecycleObserver.m

```diff
@@ -0,0 +1,115 @@
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#import <Foundation/Foundation.h>
+#import "OneSignalLifecycleObserver.h"
+#import "OneSignal.h"
+#import "OneSignalCommonDefines.h"
+#import "OneSignalTracker.h"
+#import "OneSignalLocation.h"
+#import "OSMessagingController.h"
+
+@implementation OneSignalLifecycleObserver
+
+static OneSignalLifecycleObserver* _instance = nil;
+
++(OneSignalLifecycleObserver*) sharedInstance {
+    @synchronized( _instance ) {
+        if( !_instance ) {
+            _instance = [[OneSignalLifecycleObserver alloc] init];
+        }
+    }
+    
+    return _instance;
+}
+
++ (BOOL)isAppUsingUIScene {
+    if (@available(iOS 13.0, *)) {
+        return [[NSBundle mainBundle] objectForInfoDictionaryKey:@"UIApplicationSceneManifest"] != nil;
+    }
+    return NO;
+}
+
++ (void)registerLifecycleObserver {
+    // Replacing swizzled lifecycle selectors with notification center observers for scene based Apps
+    if ([self isAppUsingUIScene]) {
+        [self registerLifecycleObserverAsUIScene];
+    } else {
+        [self registerLifecycleObserverAsUIApplication];
+    }
+}
+
++ (void)registerLifecycleObserverAsUIScene {
+    if (@available(iOS 13.0, *)) {
+        [OneSignal onesignal_Log:ONE_S_LL_INFO message:@"registering for Scene Lifecycle notifications"];
+        [[NSNotificationCenter defaultCenter] addObserver:[OneSignalLifecycleObserver sharedInstance] selector:@selector(didEnterBackground) name:UISceneDidEnterBackgroundNotification object:nil];
+        [[NSNotificationCenter defaultCenter] addObserver:[OneSignalLifecycleObserver sharedInstance] selector:@selector(didBecomeActive) name:UISceneDidActivateNotification object:nil];
+        [[NSNotificationCenter defaultCenter] addObserver:[OneSignalLifecycleObserver sharedInstance] selector:@selector(willResignActive) name:UISceneWillDeactivateNotification object:nil];
+    }
+}
+
++ (void)registerLifecycleObserverAsUIApplication {
+    [OneSignal onesignal_Log:ONE_S_LL_INFO message:@"registering for Application Lifecycle notifications"];
+    [[NSNotificationCenter defaultCenter] addObserver:[OneSignalLifecycleObserver sharedInstance] selector:@selector(didEnterBackground) name:UIApplicationDidEnterBackgroundNotification object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:[OneSignalLifecycleObserver sharedInstance] selector:@selector(didBecomeActive) name:UIApplicationDidBecomeActiveNotification object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:[OneSignalLifecycleObserver sharedInstance] selector:@selector(willResignActive) name:UIApplicationWillResignActiveNotification object:nil];
+}
+
++ (void)removeObserver {
+    [[NSNotificationCenter defaultCenter] removeObserver:[OneSignalLifecycleObserver sharedInstance]];
+}
+     
+- (void)didBecomeActive {
+    [OneSignal onesignal_Log:ONE_S_LL_INFO message:@"application/scene didBecomeActive"];
```

Comment:
Sorry poor wording on last my comment, this should be verbose or debug. The fact on travis is going over it's max log means we should change what we are setting as for the setLogLevel in the tests

Reply:
This was due to setting in the log level in the NotificationExtensionHandler